### PR TITLE
[lua] Added pathing for LB2 NMs

### LINKED ISSUE #2850: 🐛 Atop the Highest Mountains(level 55 limit break quest)
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)

## OS / platform the server is running (if known)

Window 11

## Branch affected by issue 

`base`

## Steps to reproduce
Start level 55 quest from Maat
go to Xarcabard
Those 3 NM's defending frigicite from ???'s
Boreal hound, Boreal Tiger, and Boreal Coeurl not roaming far enough for me to grab frigicite without engaging.
I remember that in retail I used to use mount to grab item without fighting.

## Expected behavior
Should able to grab frigicite without fighting any NM

### PR DESCRIPTION
**Pull request by Nemisis1987**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Closes #2850 by adding a patrol path with delays along the paths.

## Steps to test these changes

Zone into Xarcabard and go to LB2 NMs
`!gotoname Boreal_Hound`
Watch pathing of the NMs

### CODE REVIEW

**Comment by TeoTwawki** [comment]
nodes sourced from retail capture?

**Comment by Xaver-DaRed** [comment]
No. He was given a set of captures + videos after the fact, but the videos seemingly got corrupted, from what I understood.

So this positions will need updated once he gets the info again.

Reminder: Use amend + force push. We don'twant an extra commit.

**Comment by TeoTwawki** [comment]
On retail these also stop at specific point and look in multiple directions. This seems to have the stopping, but not the turning. If not comfortable trying to add that, commenting needs added about it.

**Comment by Nemisis1987** [comment]
Updated to match retail pathing behavior. Expect a new commit to add direction change while stopped.

File: scripts/zones/Xarcabard/mobs/Boreal_Coeurl.lua [hunk 1]
```diff
@@ -8,13 +8,46 @@ local ID = zones[xi.zone.XARCABARD]
...
 entity.onMobSpawn = function(mob)
     -- Failsafe to make sure NPC is down when NM is up
     if xi.settings.main.OLDSCHOOL_G2 then
         GetNPCByID(ID.npc.BOREAL_COEURL_QM):showNPC(0)
     end
 end
 
+entity.onMobRoam = function(mob)
+    local PathingIndex = mob:getLocalVar('PathingIndex')
+    local ResumePathingIndex = mob:getLocalVar('ResumePathingIndex')
```
Comment (Xaver-DaRed) [inline scripts/zones/Xarcabard/mobs/Boreal_Coeurl.lua:33]: Variable names use lowerCamelCase.
`pathingIndex`
`resumePathingIndex`

File: scripts/zones/Xarcabard/mobs/Boreal_Hound.lua [hunk 2]
```diff
@@ -15,6 +32,26 @@ entity.onMobSpawn = function(mob)
...
 
+entity.onMobRoam = function(mob)
+    local PathingIndex = mob:getLocalVar('PathingIndex')
+    local ResumePathingIndex = mob:getLocalVar('ResumePathingIndex')
+    
+    if mob:isFollowingPath() then
+        mob:setLocalVar('ResumePathingIndex', os.time() + 4) -- Make sure mob is waiting 4 seconds to path after stopping
+    elseif (os.time() > ResumePathingIndex) then
+        if (math.random() > .25) then
+            PathingIndex = PathingIndex + math.random(1,3)
```
Comment (Xaver-DaRed) [inline scripts/zones/Xarcabard/mobs/Boreal_Hound.lua:43]:
```lua
elseif os.time > resumePathingIndex then
    if math.random(1, 100) > 25 then
        pathingIndex = pathingIndex + math.random(1, 3)
```
- This is not c++. Lua doesnt need parenthesis wraping the conditions.
- Space after commas.
- Whenever possible, dont use math.random(nothing). Use concrete min and max

**Comment by Xaver-DaRed** [comment]
I would recommend you re-read the contributing guide. And concretely the lua style guide.
https://github.com/LandSandBoat/server/wiki/Development-Guide#lua

File: scripts/zones/Xarcabard/mobs/Boreal_Hound.lua (hunk 2 above)
Comment (TeoTwawki) [reply scripts/zones/Xarcabard/mobs/Boreal_Hound.lua:43]: why are we even randoming? retail is pretty consistent about what every step of the path including the directions faced give or take a little latency on when the updated positions reach the client via packet.

**Comment by TeoTwawki** [comment]
superseded by https://github.com/LandSandBoat/server/pull/5457 but I hope you aren't discouraged from future controbutions

**Pull request closed without merging**